### PR TITLE
Make InteractUsing async, make tools use DoAfter.

### PR DESCRIPTION
**Pull request by Zumorica**

### CODE REVIEW

File: Content.Server/GameObjects/Components/PlaceableSurfaceComponent.cs [hunk 1]
```diff
@@ -19,7 +20,9 @@ public override void ExposeData(ObjectSerializer serializer)
             serializer.DataField(ref _isPlaceable, "IsPlaceable", true);
         }
 
-        public bool InteractUsing(InteractUsingEventArgs eventArgs)
+#pragma warning disable 1998
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/PlaceableSurfaceComponent.cs:23]: So you CAN avoid this warning by not marking the method in question as `async` and then using `Task.FromResult()` to wrap the return value buuuut screw that noise so I'd be open to just doing a project wide muting of this warning.

Comment (Zumorica) [reply Content.Server/GameObjects/Components/PlaceableSurfaceComponent.cs:25]: Agreed, I added `<NoWarn>1998</NoWarn>` to the project and removed all added pragmas.

File: Content.Server/GameObjects/Components/AnchorableComponent.cs [hunk 2]
```diff
@@ -24,9 +25,9 @@ public class AnchorableComponent : Component, IInteractUsing
         /// <param name="collidable">The physics component of the owning entity</param>
```
Comment (DrSmugleaf) [inline Content.Server/GameObjects/Components/AnchorableComponent.cs:25]: This summary needs to be updated.

Comment (Zumorica) [reply Content.Server/GameObjects/Components/AnchorableComponent.cs:25]:
```suggestion
        /// <param name="collidable">The collidable component of the owning entity</param>
```

**Pull request merged**